clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1196,16 +1196,24 @@ private long maybeDrainPendingCalls(long now) {
             long pollTimeout = Long.MAX_VALUE;
             log.trace("Trying to choose nodes for {} at {}", pendingCalls, now);
 
-            Iterator<Call> pendingIter = pendingCalls.iterator();
-            while (pendingIter.hasNext()) {
-                Call call = pendingIter.next();
+            List<Call> toRemove = new ArrayList<>();
+            int pendingSize = pendingCalls.size();
+            for (int i = 0; i < pendingSize; i++) {
```

Comment:
Please add comments to explain why we use this for-loop style

Reply:
Yes, updated it. Thank you.